### DISCUSSION TITLE
Preference Towards `canvas` Rendering Over SVG

### DISCUSSION THREAD

**Discussion by Brandon-Strong-Abbott** [wojtekmaj/react-pdf#1222] [Q&A] [answered]
Why would the `renderMode` be better defaulted to `"canvas"` instead of `"svg"`? I was using this library as well as [`react-to-print`](https://github.com/gregnb/react-to-print) to display and print PDF documents and found that the Safari browser [does not adequately clear out `canvas` data](https://bugs.webkit.org/show_bug.cgi?id=195325) to allow a document larger than 4 pages to be rendered to the print preview on iOS. This was solved by simply switching to `svg` for `renderMode`, but I noticed this is not the preferred method, is no longer maintained, and may not exist in future versions. I'm sure there is a great reason for this default as well as the wanting to move away from `svg` rendering and was simply curious as to why.

**Comment by wojtekmaj** [comment, marked as answer]
Because SVG mode is considered experimental, has plenty of bugs, and thus is not yet officially supported by Mozilla. Still, a cool option for simple use cases, so I gave the users the ability to opt in, if they're sure what they're doing.

**Reply by Brandon-Strong-Abbott** [reply]
Thanks for the timely response and maintaining this library! Do you have any specifics/resources around the bugs mentioned?

**Reply by wojtekmaj** [reply]
https://github.com/mozilla/pdf.js/issues?q=is%3Aissue+label%3Asvg

They sometimes reply that they won't fix it now because SVG is experimental, like here: https://github.com/mozilla/pdf.js/issues/11811#issuecomment-614290087
